content/docs/guides/tls-encryption.md

```diff
@@ -0,0 +1,107 @@
+---
+title: TLS encryption
+sort_rank: 1
+---
+
+# Securing Prometheus API and UI endpoints using TLS encryption
+
+Prometheus does not directly support [Transport Layer Security](https://en.wikipedia.org/wiki/Transport_Layer_Security) (TLS) encryption for connections to Prometheus instances (i.e. to the expression browser or [HTTP API](../../prometheus/latest/querying/api)). If you would like to enforce TLS for those connections, we recommend using Prometheus in conjunction with a [reverse proxy](https://www.nginx.com/resources/glossary/reverse-proxy-server/) and applying TLS at the proxy layer. You can use any reverse proxy you like with Prometheus, but in this cookbook we'll provide an [nginx example](#nginx-example).
+
+NOTE: Although TLS connections *to* Prometheus instances are not supported, TLS is supported for connections *from* Prometheus instances to [scrape targets](../prometheus/latest/configuration/configuration/#<tls_config>).
+
+## nginx example
+
+Let's say that you want to run a Prometheus instance behind an [nginx](https://www.nginx.com/) server available at the `my-proxy.com` domain (which you own), and for all Prometheus endpoints to be available via the `/prometheus` endpoint. The full URL for Prometheus' `/metrics` endpoint would thus be:
+
+```
+https://my-proxy.com/prometheus/metrics
+```
+
+Let's also say that you've generated the following using [OpenSSL](https://www.digitalocean.com/community/tutorials/openssl-essentials-working-with-ssl-certificates-private-keys-and-csrs) or an analogous tool:
+
+* a self-signed SSL certificate at `/root/certs/my-proxy.com/my-proxy.com.crt`
+* an SSL key at `/root/certs/my-proxy.com/my-proxy.com.key`
+
+You can generate the above certificate and key using this command:
+
+```bash
+mkdir -p /root/certs/my-proxy.com && cd /root/certs/my-proxy.com
+openssl req \
+  -x509 \
+  -newkey rsa:4096 \
+  -nodes \
+  -keyout my-proxy.com.key \
+  -out my-proxy.com.crt
+```
+
+Fill out the appropriate information at the prompts, and make sure to enter `my-proxy.com` at the `Common Name` prompt.
+
+## nginx configuration
+
+Below is an example [`nginx.conf`](https://www.nginx.com/resources/wiki/start/topics/examples/full/) configuration file, located at the default path of `/usr/local/etc/nginx/nginx.conf`. With this configuration, nginx will:
+
+* enforce TLS encryption using your provided certificate and key
+* proxy all connections to the `/prometheus` endpoint to a Prometheus server running on the same host (while removing the `/prometheus` from the URL)
+
+```conf
+events {}
+
+http {
+    server {
+        listen              443 ssl;
+        server_name         my-proxy.com;
+        ssl_certificate     /root/certs/my-proxy.com/my-proxy.com.crt;
+        ssl_certificate_key /root/certs/my-proxy.com/my-proxy.com.key;
+
+        location /prometheus {
+          proxy_pass http://localhost:9090/;
+        }
+    }
+}
+```
+
+To start nginx:
```

Comment:
You're using 443, so should need to mention that this needs to run as root, and it'd be wise to add the flag pointing to the config file